### DISCUSSION TITLE
Getting Postgres to work with Blitz

### DISCUSSION THREAD

**Discussion by Kay2dan** [blitz-js/blitz#1101] [Q&A] [answered]
Hi, after having created some content on the frontend, now I am paying attention to the DB & Prisma. 

Here is the list of actions I have already done:

- Installed Postgres 10.14 on Mint 19.04 OS.
- Created user with permissions to create roles & create Db.
- Created OS level user (aim_high) by the same name as the user in Postgres.
- I also followed this [link](https://stackoverflow.com/a/18664239/434697) to change postgres authentication from `peer` to `md5`.
- Updated `prisma.schema` to refer to env.local fine which contains the line: `DATABASE_URL=postgresql://aim_high:123@localhost:5432/aim_high `
- Ran `blitz start`: `localhost:3000` looks fine (can see my pages & content in browser)
- Opened `localhost:5432/aim_high`, error displayed: connection was reset...
- Ran `blitz generate all blahblah text`, Model & files created successfully; verified by looking at prisma.schema & folder
- Ran `blitz db migrate`, error: `P1000: Authentication failed against database server at `localhost`, the provided database credentials for `aim_high` are not valid.`
- Ran `blitz studio`, error: 
`(node:22297) TypeError Plugin: @blitzjs/cli: Cannot read property 'dim' of undefined
module: @oclif/config@1.15.1
task: runHook command_not_found
plugin: @blitzjs/cli
root: /home/sk/www/aim-high/node_modules/blitz/node_modules/@blitzjs/cli`

Would appreciate help please? Why is `localhost:5432` server not working? I can log into the aim_high postgres user & access psql db by the same name (aim_high), however, Blitz is unable to connect to it. I suspect the authentication settings above might have something to do with it, however, I have tried different options, to no success.

**Comment by Kay2dan** [comment, marked as answer]
I managed to get the authentication error resolved by changing the postgres authentication from md5 to trusted for ip6, ip4 etc.

Now, `blitz db migrate` worked & I am able to `blitz console` & get the records for blahblah above.
However, the localhost:5432 still doesnt load (because of no backend server like Express.js?).

And `blitz studio` was missing the `db` attribute, so it runs now & opens the page `http://localhost:5555`, however, I get the error `that failed spectacularly`.

**Reply by flybayer** [reply]
> Now, blitz db migrate worked & I am able to blitz console & get the records for blahblah above.
However, the localhost:5432 still doesnt load (because of no backend server like Express.js?).

What do you expect to see at `localhost:5432`? 

> And blitz studio was missing the db attribute, so it runs now & opens the page http://localhost:5555, however, I get the error that failed spectacularly.

What error do you get in the console for this?

**Reply by Kay2dan** [reply]
>> What do you expect to see at `localhost:5432`?

I see what you mean.  The postgres db is accessed via the Prisma studio. I was thinking I would get some GUI for postgres at port 5432 or some message to say that a server is running on this port, which is not how it works, I imagine.
 
>> What error do you get in the console for this?

The error on the screen itself says: `that failed spectacularly`. Screen:

![Screenshot from 2020-09-19 08-01-26](https://user-images.githubusercontent.com/6276091/93661330-d10e7480-fa4e-11ea-8f54-4159ba637d4f.jpg)

I am going to look into this a bit more & hopefully resolve this. Its not blocking me from continuing on so its fine for now.

Thank you Brandon for helping out with these newbie questions. I am much grateful.

**Reply by flybayer** [reply]
Ok! :) Usually when you see `that failed spectacularly.` on the studio web page, there will be an error saying what went wrong in the terminal (where you ran `blitz db studio`).

**Reply by Kay2dan** [reply]
I'm a little embarrassed to say that I just ran the `blitz db studio` command and it worked. I cannot recall having changed anything and yet there it is on the screen. 

![Selection_002](https://user-images.githubusercontent.com/6276091/93686407-ed8dc980-faad-11ea-9d27-5f5dabe5dc64.png)

Thank you again for the help Brandon! Really appreciate it.
